src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.cs

```diff
@@ -220,7 +220,7 @@ public WebSocketsTransport(WebSocketOptions options, IDuplexPipe application, De
                             {
                                 Log.SendPayload(_logger, buffer.Length);
 
-                                var webSocketMessageType = (_connection.TransferMode == TransferMode.Binary
```

Comment:
Do this once on creation too?

Reply:
I actually wanted to raise that. So technically right now, the application could **change** the transfer format mid-stream. This would be somewhat legal, but I guess it would also be weird because you have no guarantee which messages got flushed to the socket before they were sent... so in typing this comment I just convinced myself it needs to be captured once :). I'll make that change.